### ISSUE TITLE
Can't create a service with v1.0rc2 => Timeout: request did not complete within allowed duration

### ISSUE THREAD

**Issue by djannot** [kubernetes/kubernetes#11304]
s3picsService.json:

```json
{
   "kind":"Service",
   "apiVersion":"v1",
   "metadata":{
      "name":"s3pics",
      "labels":{
         "name":"s3pics"
      }
   },
   "spec":{
      "ports": [
        {
          "port":80,
          "targetPort":8080,
          "protocol":"TCP"
        }
      ],
      "selector":{
         "name":"s3pics"
      }
   }
}
```

kubernetes/platforms/linux/amd64/kubectl --server=10.64.231.42:8888 create --v=9 -f s3picsService.json       
I0715 15:06:11.266410   29402 decoder.go:141] decoding stream as JSON
I0715 15:06:11.267864   29402 debugging.go:101] curl -k -v -XGET  -H "User-Agent: kubectl/v1.0.0 (linux/amd64) kubernetes/cd82144" http://10.64.231.42:8888/api
I0715 15:06:11.269408   29402 debugging.go:120] GET http://10.64.231.42:8888/api 200 OK in 1 milliseconds
I0715 15:06:11.269665   29402 debugging.go:126] Response Headers:
I0715 15:06:11.269828   29402 debugging.go:129]     Date: Wed, 15 Jul 2015 13:06:11 GMT
I0715 15:06:11.269992   29402 debugging.go:129]     Content-Length: 32
I0715 15:06:11.270104   29402 debugging.go:129]     Content-Type: application/json
I0715 15:06:11.270285   29402 request.go:779] Response Body: {
  "versions": [
    "v1"
  ]
}
I0715 15:06:11.273093   29402 request.go:482] Request Body: {"kind":"Service","apiVersion":"v1","metadata":{"name":"s3pics","namespace":"default","creationTimestamp":null,"labels":{"name":"s3pics"}},"spec":{"ports":[{"protocol":"TCP","port":80,"targetPort":8080,"nodePort":0}],"selector":{"name":"s3pics"},"type":"ClusterIP","sessionAffinity":"None"},"status":{"loadBalancer":{}}}
I0715 15:06:11.273309   29402 debugging.go:101] curl -k -v -XPOST  -H "User-Agent: kubectl/v1.0.0 (linux/amd64) kubernetes/cd82144" http://10.64.231.42:8888/api/v1/namespaces/default/services
I0715 15:08:11.278042   29402 debugging.go:120] POST http://10.64.231.42:8888/api/v1/namespaces/default/services 504 Gateway Timeout in 120004 milliseconds
I0715 15:08:11.278494   29402 debugging.go:126] Response Headers:
I0715 15:08:11.278552   29402 debugging.go:129]     Content-Type: application/json
I0715 15:08:11.278579   29402 debugging.go:129]     Date: Wed, 15 Jul 2015 13:08:11 GMT
I0715 15:08:11.278601   29402 debugging.go:129]     Content-Length: 214
I0715 15:08:11.278686   29402 request.go:779] Response Body: {
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "Timeout: request did not complete within allowed duration",
  "reason": "Timeout",
  "details": {},
  "code": 504
}
I0715 15:08:11.279406   29402 helpers.go:170] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "error when creating \"s3picsService.json\": Timeout: request did not complete within allowed duration",
  "reason": "Timeout",
  "details": {},
  "code": 504
}]
F0715 15:08:11.279552   29402 helpers.go:72] Error from server: error when creating "s3picsService.json": Timeout: request did not complete within allowed duration

**Comment by erictune** [comment]
What Cloud provider are you using?

**Comment by erictune** [comment]
assigned to @lavalamp to delegate to someone who understands apiserver well enough to support this.

**Comment by djannot** [comment]
I'm running Kubernetes on a CoreOS cluster.

**Comment by erictune** [comment]
Can you check and/or provide apiserver logs (e.g. using `journalctl`)  from the same time frame as the failure?

**Comment by lavalamp** [comment]
Is etcd healthy? apiserver logs would be very helpful.

**Comment by djannot** [comment]
I've just found that the service has now been created.

And I can delete and recreate quickly.

I don't know if the issue was that I've tried to create the service just after I've deployed k8s. Perhaps it needed some time to be operational.

I've seen a lot of these errors in the logs, but don't see them anymore:
Jul 15 16:24:33 coreos2 kube-apiserver[2575]: E0715 16:24:33.384935    2575 repair.go:52] unable to persist the updated port alloc
ations: servicenodeportallocation "" cannot be updated: the provided resource version does not match
Jul 15 16:25:48 coreos2 kube-apiserver[2575]: E0715 16:25:48.270683    2575 repair.go:67] unable to persist the updated service IP
 allocations: serviceipallocation "" cannot be updated: the provided resource version does not match
